### PR TITLE
[Python API] Add missing opest8 ops to compatibility python API

### PR DESCRIPTION
**Pull request by mitruska**
### Details:
As opset8 has been restored in compatibility API (https://github.com/openvinotoolkit/openvino/pull/8625)
missed ops that are only in new API need to be added also to the compatibility API

- [x] Slice-8 (already added to new API: https://github.com/openvinotoolkit/openvino/pull/7965)
(+ add missing optional name attribute)

- [x] GatherND-8 (already added to new API: https://github.com/openvinotoolkit/openvino/pull/8374)

### Tickets:
 - Related to 68456, 54654, 66661

### CODE REVIEW

**Review by tomdol — APPROVED** [review]

**Review by artkuli — APPROVED** [review]

**Review by achetver — APPROVED** [review]

File: runtime/bindings/python/src/compatibility/ngraph/opset8/ops.py [hunk 1]
```diff
@@ -367,3 +367,53 @@ def random_uniform(
...
+
+
+@nameable_op
+def slice(
+        data: NodeInput,
+        start: NodeInput,
+        stop: NodeInput,
+        step: NodeInput,
+        axes: NodeInput = None,
+        name: Optional[str] = None,
```
Comment (jiwaszki) [inline runtime/bindings/python/src/compatibility/ngraph/opset8/ops.py:379]: `axes` can be Optional as well.

File: runtime/bindings/python/src/openvino/opset8/ops.py [hunk 2]
```diff
@@ -375,7 +375,8 @@ def slice(
         start: NodeInput,
         stop: NodeInput,
         step: NodeInput,
-        axes: NodeInput = None
+        axes: NodeInput = None,
```
Comment (jiwaszki) [inline runtime/bindings/python/src/openvino/opset8/ops.py:378]: Same here.

File: runtime/bindings/python/src/compatibility/ngraph/opset8/ops.py (hunk 1 above)
Comment (mitruska) [reply runtime/bindings/python/src/compatibility/ngraph/opset8/ops.py]: Sure, done

File: runtime/bindings/python/src/openvino/opset8/ops.py (hunk 2 above)
Comment (mitruska) [reply runtime/bindings/python/src/openvino/opset8/ops.py]: Done

**Review by jiwaszki — APPROVED** [review]
🚀 🚀 🚀

**Review by ilyachur — APPROVED** [review]

**Comment by ilyachur** [comment]
@itikhono FYI

**Pull request merged**